biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -232,62 +239,86 @@ subroutine LoggingMortality_frac( pft_i, dbh, canopy_layer, lmort_direct, &
       ! Local variables
       integer :: cur_harvest_tag ! the harvest tag of the cohort today
       real(r8) :: harvest_rate ! the final harvest rate to apply to this cohort today
+      real(r8) :: state_vector(n_landuse_cats)
+      logical  :: site_secondaryland_first_exceeding_min
 
       ! todo: probably lower the dbhmin default value to 30 cm
       ! todo: change the default logging_event_code to 1 september (-244)
       ! todo: change the default logging_direct_frac to 1.0 for cmip inputs
       ! todo: check outputs against the LUH2 carbon data
       ! todo: eventually set up distinct harvest practices, each with a set of input paramaeters
       ! todo: implement harvested carbon inputs
-      
-      if (logging_time) then 
 
-         ! Pass logging rates to cohort level 
-         
-         if (hlm_use_lu_harvest == ifalse) then
-            ! 0=use fates logging parameters directly when logging_time == .true.
-            ! this means harvest the whole cohort area
-            harvest_rate = 1._r8
-            
-         else if (hlm_use_lu_harvest == itrue .and. hlm_harvest_units == hlm_harvest_area_fraction) then
-            ! We are harvesting based on areal fraction, not carbon/biomass terms. 
-            ! 1=use area fraction from hlm
-            ! combine forest and non-forest fracs and then apply:
-            ! primary and secondary area fractions to the logging rates, which are fates parameters
-            
-            ! Definitions of the underlying harvest land category variables
-            ! these are hardcoded to match the LUH input data via landuse.timseries file (see dynHarvestMod)
-            ! these are fractions of vegetated area harvested, split into five land category variables
-            ! HARVEST_VH1 = harvest from primary forest
-            ! HARVEST_VH2 = harvest from primary non-forest
-            ! HARVEST_SH1 = harvest from secondary mature forest
-            ! HARVEST_SH2 = harvest from secondary young forest
-            ! HARVEST_SH3 = harvest from secondary non-forest (assume this is young for biomass)
-
-            ! Get the area-based harvest rates based on info passed to FATES from the boundary condition
-            call get_harvest_rate_area (patch_land_use_label, hlm_harvest_catnames, &
-                 hlm_harvest_rates, frac_site_primary, secondary_age, harvest_rate)
+      call get_luh_statedata(bc_in, state_vector)
+      site_secondaryland_first_exceeding_min =  (state_vector(secondaryland) .gt. currentSite%min_allowed_landuse_fraction) &
+           .and. (.not. currentSite%landuse_vector_gt_min(secondaryland))
+
+      if (.not. currentSite%transition_landuse_from_off_to_on) then
```

Comment:
Why is this comparison made? I don't think I understand this flag.

Reply:
The `transition_landuse_from_off_to_on` is for handling the special case of the first timestep after leaving potential vegetation mode. In this case, all prior historical land-use, including harvest, needs to be applied on that first day. So logging rates on that day are what is required to deforest exactly the amount of primary lands that will give the amount of secondary lands dictated by the land use state vector for that year, rather than whatever the continuous logging rate for that year is supposed to be according to the transition matrix.  

Will note to add comments in code to that effect.

Reply:
thanks for the explanation